agent/jvm/src/main/java/io/mockk/proxy/jvm/dispatcher/JvmMockKWeakMap.java

```diff
@@ -9,30 +9,30 @@
 import java.util.concurrent.ConcurrentHashMap;
 
 public class JvmMockKWeakMap<K, V> implements Map<K, V> {
-    private final Map<Object, V> map = new ConcurrentHashMap<Object, V>();
+    private final Map<Object, V> target = new ConcurrentHashMap<Object, V>();
```

Comment:
Nit* Renaming variable seems unnecessary even if `getTarget` is the desired name for the method which returns the backing field

Reply:
Well I have based on a naming that is applied in mockito. Having a map with inner property map sounds strange. Target or innerMap sound more appropriate

Reply:
Please check the latest commit. I have introduced  MockHandlerMap (Map with isMock method), just to decouple map implementation. Maybe I have misunderstood your suggestion, this way baseAdvise delegates isMock to inner map

Reply:
Tests results 
<img width="1678" alt="Screenshot 2020-08-04 at 22 30 15" src="https://user-images.githubusercontent.com/10225385/89375319-45a97180-d6a2-11ea-85c7-b476d777d295.png">